In [1]:
from google.colab import drive
drive.mount('/content/drive')

#Optional: move to the desired location:
#%cd drive/My Drive/DIRECTORY_IN_YOUR_DRIVE

Mounted at /content/drive


In [2]:
import pandas as pd
import os

In [3]:
data_folder = 'drive/My Drive/Omdena/SriLanka/Autism/data'
csv_files = os.listdir(data_folder)
print(csv_files)


['Autism Spectrum Disorder Screening Data for Toddlers in Saudi Arabia Data Set.csv', 'Autism-Child-Data.csv', 'data_csv.csv', 'Toddler Autism dataset July 2018.csv']


# Saudi Data

In [16]:
saudi_df = pd.read_csv(os.path.join(data_folder,'Autism Spectrum Disorder Screening Data for Toddlers in Saudi Arabia Data Set.csv'))

In [17]:
saudi_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   A10                             506 non-null    int64 
 1   A9                              506 non-null    int64 
 2   A8                              506 non-null    int64 
 3   A7                              506 non-null    int64 
 4   A6                              506 non-null    int64 
 5   A5                              506 non-null    int64 
 6   A4                              506 non-null    int64 
 7   A3                              506 non-null    int64 
 8   A2                              506 non-null    int64 
 9   A1                              506 non-null    int64 
 10  Region                          506 non-null    object
 11  Family member with ASD history  506 non-null    object
 12  Who is completing the test      506 non-null    ob

In [41]:
saudi_df_updated = saudi_df[['A1', 'A2','A3','A4','A5','A6','A7','A8','A9','A10', 'Class']]
saudi_df_updated['src'] = 'Saudi'

<ipython-input-41-52aab8eef489>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  saudi_df_updated['src'] = 'Saudi'


In [42]:
#change columns for consistency
saudi_df_updated.head()

A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Class    src
0   0   0   0   0   0   1   1   1   0    0      0  Saudi
1   0   0   1   0   1   0   0   1   0    0      0  Saudi
2   0   0   0   0   0   0   1   0   0    0      0  Saudi
3   0   0   0   0   0   0   0   0   0    0      0  Saudi
4   0   0   0   0   0   0   0   0   0    0      0  Saudi

#UCI Data

In [19]:
uci_child_df = pd.read_csv(os.path.join(data_folder, 'Autism-Child-Data.csv'))

In [20]:
uci_child_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 292 entries, 0 to 291
Data columns (total 21 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   A1_Score         292 non-null    int64 
 1   A2_Score         292 non-null    int64 
 2   A3_Score         292 non-null    int64 
 3   A4_Score         292 non-null    int64 
 4   A5_Score         292 non-null    int64 
 5   A6_Score         292 non-null    int64 
 6   A7_Score         292 non-null    int64 
 7   A8_Score         292 non-null    int64 
 8   A9_Score         292 non-null    int64 
 9   A10_Score        292 non-null    int64 
 10  age              292 non-null    object
 11  gender           292 non-null    object
 12  ethnicity        292 non-null    object
 13  jundice          292 non-null    object
 14  austim           292 non-null    object
 15  contry_of_res    292 non-null    object
 16  used_app_before  292 non-null    object
 17  result           292 non-null    in

In [21]:
uci_child_df_updated = uci_child_df[['A1_Score','A2_Score','A3_Score','A4_Score','A5_Score','A6_Score','A7_Score','A8_Score','A9_Score','A10_Score', 'Class/ASD']]
uci_child_df_updated['Class'] = uci_child_df_updated['Class/ASD'].apply(lambda x: 1 if x == 'YES' else 0)
uci_child_df_updated.drop(columns = ['Class/ASD'], inplace = True)


<ipython-input-21-7e97ecb66d60>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  uci_child_df_updated['Class'] = uci_child_df_updated['Class/ASD'].apply(lambda x: 1 if x == 'YES' else 0)
/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:4906: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [43]:
uci_child_df_updated.rename(columns={"A1_Score": "A1", "A10_Score": "A10","A2_Score": "A2","A3_Score": "A3","A4_Score": "A4","A5_Score": "A5","A6_Score": "A6","A7_Score": "A7","A8_Score": "A8","A9_Score": "A9",}, inplace = True)
uci_child_df_updated['src'] = 'UCI_child'

/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:5039: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(
<ipython-input-43-984a2471f773>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  uci_child_df_updated['src'] = 'UCI_child'


In [44]:
uci_child_df_updated.head(5)

A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Class        src
0   1   1   0   0   1   1   0   1   0    0      0  UCI_child
1   1   1   0   0   1   1   0   1   0    0      0  UCI_child
2   1   1   0   0   0   1   1   1   0    0      0  UCI_child
3   0   1   0   0   1   1   0   0   0    1      0  UCI_child
4   1   1   1   1   1   1   1   1   1    1      1  UCI_child

#Thabtah Toddler

In [24]:
thabtah_df = pd.read_csv(os.path.join(data_folder, 'Toddler Autism dataset July 2018.csv'))

In [25]:
thabtah_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1054 entries, 0 to 1053
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Case_No                 1054 non-null   int64 
 1   A1                      1054 non-null   int64 
 2   A2                      1054 non-null   int64 
 3   A3                      1054 non-null   int64 
 4   A4                      1054 non-null   int64 
 5   A5                      1054 non-null   int64 
 6   A6                      1054 non-null   int64 
 7   A7                      1054 non-null   int64 
 8   A8                      1054 non-null   int64 
 9   A9                      1054 non-null   int64 
 10  A10                     1054 non-null   int64 
 11  Age_Mons                1054 non-null   int64 
 12  Qchat-10-Score          1054 non-null   int64 
 13  Sex                     1054 non-null   object
 14  Ethnicity               1054 non-null   object
 15  Jaun

In [45]:
thabtah_df_updated = thabtah_df[['A1', 'A2','A3','A4','A5','A6','A7','A8','A9','A10',"Class"]]
thabtah_df_updated['Class'] = thabtah_df_updated['Class'].apply(lambda x: 1 if x == 'Yes' else 0)
#thabtah_df_updated.drop(columns = ['Class/ASD'], inplace = True)
thabtah_df_updated['src'] = 'thabtah_toddler'

<ipython-input-45-ecdb9736bcd1>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  thabtah_df_updated['Class'] = thabtah_df_updated['Class'].apply(lambda x: 1 if x == 'Yes' else 0)
<ipython-input-45-ecdb9736bcd1>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  thabtah_df_updated['src'] = 'thabtah_toddler'


In [46]:
thabtah_df_updated.head()

A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Class              src
0   0   0   0   0   0   0   1   1   0    1      0  thabtah_toddler
1   1   1   0   0   0   1   1   0   0    0      1  thabtah_toddler
2   1   0   0   0   0   0   1   1   0    1      1  thabtah_toddler
3   1   1   1   1   1   1   1   1   1    1      1  thabtah_toddler
4   1   1   0   1   1   1   1   1   1    1      1  thabtah_toddler

#Github Data

In [31]:
github_df = pd.read_csv(os.path.join(data_folder, 'data_csv.csv'))

In [32]:
github_df_child = github_df[github_df.Age_Years < 13]

In [33]:
github_df_child.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1206 entries, 0 to 1984
Data columns (total 28 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   CASE_NO_PATIENT'S                                   1206 non-null   int64  
 1   A1                                                  1206 non-null   int64  
 2   A2                                                  1206 non-null   int64  
 3   A3                                                  1206 non-null   int64  
 4   A4                                                  1206 non-null   int64  
 5   A5                                                  1206 non-null   int64  
 6   A6                                                  1206 non-null   int64  
 7   A7                                                  1206 non-null   int64  
 8   A8                                                  1206 non-null   int64  
 9

In [52]:
#saving the child data to check for all 0 (from a1 to a9) errors
github_df_child.to_csv('drive/My Drive/Omdena/SriLanka/Autism/github_child.csv', index = False)

After checking the github data for children, it seems the last ~500 rows have 0 values for a1 to a9 even though some of those datapoints are considered ASD positive(class = Yes). So, those rows will be deleted and then merged with other data. As such out of 1206 datapoints, the first 714 datapoint (that are found to be not-erroneous) are taken

In [55]:
github_df_child_ = github_df_child[:714].copy()

In [56]:
github_df_child_updated = github_df_child_[['A1', 'A2','A3','A4','A5','A6','A7','A8','A9','A10_Autism_Spectrum_Quotient',"ASD_traits"]]
github_df_child_updated['Class'] = github_df_child_updated['ASD_traits'].apply(lambda x: 1 if x == 'Yes' else 0)
github_df_child_updated.drop(columns = ['ASD_traits'], inplace = True)

<ipython-input-56-16b52b64edbb>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  github_df_child_updated['Class'] = github_df_child_updated['ASD_traits'].apply(lambda x: 1 if x == 'Yes' else 0)
/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:4906: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [57]:
github_df_child_updated.rename(columns={ "A10_Autism_Spectrum_Quotient": "A10"}, inplace = True)
github_df_child_updated['src'] = 'github'

/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:5039: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(
<ipython-input-57-3a732c085085>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  github_df_child_updated['src'] = 'github'


In [58]:
github_df_child_updated.head()

A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Class     src
0   0   0   0   0   0   0   1   1   0    1      0  github
1   1   1   0   0   0   1   1   0   0    0      1  github
2   1   0   0   0   0   0   1   1   0    1      1  github
3   1   1   1   1   1   1   1   1   1    1      1  github
4   1   1   0   1   1   1   1   1   1    1      1  github

In [59]:
github_df_child_updated.shape

(714, 12)

#Combine

In [60]:
#combine all datasets
final_df = pd.concat([saudi_df_updated,uci_child_df_updated,thabtah_df_updated, github_df_child_updated], axis=0)

In [61]:
final_df.shape

(2566, 12)

In [62]:
final_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2566 entries, 0 to 1053
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   A1      2566 non-null   int64 
 1   A2      2566 non-null   int64 
 2   A3      2566 non-null   int64 
 3   A4      2566 non-null   int64 
 4   A5      2566 non-null   int64 
 5   A6      2566 non-null   int64 
 6   A7      2566 non-null   int64 
 7   A8      2566 non-null   int64 
 8   A9      2566 non-null   int64 
 9   A10     2566 non-null   int64 
 10  Class   2566 non-null   int64 
 11  src     2566 non-null   object
dtypes: int64(11), object(1)
memory usage: 260.6+ KB


In [63]:
final_df.to_csv('drive/My Drive/Omdena/SriLanka/Autism/merged_saudi_github_thabtah_toddler_child.csv',index= False)